irohad/main/impl/pg_connection_init.hpp

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_PG_CONNECTION_INIT_HPP
+#define IROHA_PG_CONNECTION_INIT_HPP
+
+#include <soci/soci.h>
+
+#include <soci/callbacks.h>
+#include <soci/postgresql/soci-postgresql.h>
+#include <boost/algorithm/string.hpp>
+#include <boost/range/algorithm/replace_if.hpp>
+
+#include "ametsuchi/impl/failover_callback_factory.hpp"
+#include "ametsuchi/impl/pool_wrapper.hpp"
+#include "ametsuchi/impl/postgres_command_executor.hpp"
+#include "ametsuchi/impl/postgres_options.hpp"
+#include "ametsuchi/reconnection_strategy.hpp"
+#include "common/result.hpp"
+#include "interfaces/permissions.hpp"
+#include "logger/logger_manager.hpp"
+
+namespace iroha {
+  namespace ametsuchi {
+    inline std::string formatPostgresMessage(const char *message) {
+      std::string formatted_message(message);
+      boost::replace_if(formatted_message, boost::is_any_of("\r\n"), ' ');
+      return formatted_message;
+    }
+
+    inline void processPqNotice(void *arg, const char *message) {
+      auto *log = reinterpret_cast<logger::Logger *>(arg);
```

Comment:
Since you use `logger::Logger`, you should include `logger/logger.hpp`.
But I would suggest to move this to cpp file and to ensure that it has this include, because otherwise the CMake `logger` target dependency must be manually added to every file using this header.
If you choose to follow my advise, please include `#include "logger/logger_fwd.hpp"` to this file.